File: backend/src/main/java/ai/giskard/utils/FileUtils.java
package ai.giskard.utils;

public class FileUtils {

    private FileUtils() {
        throw new IllegalStateException("Util file");
    }

    public static String getFileName(String name, String extension, boolean sample) {
        StringBuilder builder = new StringBuilder(name);
        if (sample) {
            builder.append(".sample");
        }

        return builder.append(".").append(extension).toString();
    }

}

File: giskard/utils/file_utils.py
from pathlib import Path
from uuid import UUID

from giskard.settings import settings


def get_file_name(name: str, extension: str, sample: bool):
    return f"{name}.sample.{extension}" if sample else f"{name}.{extension}"


def job_logs_path(job_id: UUID) -> Path:
    return settings.home_dir / "run" / "jobs" / (str(job_id) + ".log")
